middleware/lbutils.py

```diff
@@ -20,18 +20,40 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+import ast
+import contextlib
+import io
 import sys
 
 
+def post_process_list_apps(raw_output):
+    app_list = []
+    for line in raw_output.splitlines():
+        line = line.strip()
+        if line.startswith("[{") and line.endswith("}]"):
+            data = ast.literal_eval(line)
+            if isinstance(data, list) and all(
+                    isinstance(app_dict, dict) for app_dict in data):
+                for app_dict in data:
+                    app_list.append(app_dict["name"])
+                break
+    if app_list:
+        return "\n".join([f"Installed app: {app}" for app
```

Comment:
What does the prefix `Installed app: ` serve the purpose of?

Reply:
That exists so we can `grep` only the lines of interest in the `setup` script. For example, if the device has no apps installed, a `No apps to remove.` message is displayed (and that's not an error), and we need to make a distinction between a valid list of apps and any other output.

Also, thinking on the use case of `lbutils` as a standalone tool, not only as part of the onboard routine, it felt a bit awkward to just output the list of apps (or nothing if no apps are installed) with no additional context.

What do you think?

Reply:
Good point. Two things:
1. One way of addressing the need for grepping is just removing any useless output in the preprocessing.
2. The awkwardness I guess depends on what the user expects. If you're using a "list apps" command-line utility, and it literally just outputs a list of names, then I don't see a problem (this is actually the behavior of many UNIX-like command-line utilities).
All of this said, up to you.